File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}

        for num in nums:
            freq[num] = 1 + freq.get(num, 0)
        
        bucket = []

        for num, cnt in freq.items():
            bucket.append([cnt, num])
        
        bucket.sort()

        res = []

        while len(res) < k:
            res.append(bucket.pop()[1])
        
        return res

File: Data Structures & Algorithms/house-robber/submission-3.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 0) return 0;
        if (nums.length == 1) return nums[0];

        int n = nums.length;
        int[] db = new int[n];
        db[0] = nums[0];
        db[1] = Math.max(db[0], nums[1]);
        
        for (int i = 2; i<n; i++) {
            db[i] = Math.max(nums[i] + db[i-2], db[i-1]);
        }

        return Math.max(db[n-1], db[n-2]);
    }
}
